### PR TITLE
EmptyLoopBodyFixer - introduction

### PR DESCRIPTION
**Pull request by SpacePossum**
```
possum@aquarium:/home/seb/work/PHP-CS-Fixer$ php php-cs-fixer describe empty_loop_body
Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (fallback through xdebug.client_host/xdebug.client_port) :-(
Description of empty_loop_body rule.
Empty loop-body must be in configured style.

Fixer is configurable using following option:
* style ('braces', 'semicolon'): style of empty loop-bodies; defaults to 'braces'

Fixing examples:
 * Example #1. Fixing with the default configuration.
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -1,1 +1,1 @@
   -<?php while(foo());
   +<?php while(foo()){}
   
   ----------- end diff -----------

 * Example #2. Fixing with configuration: ['style' => 'semicolon'].
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -1,1 +1,1 @@
   -<?php while(foo()){}
   +<?php while(foo());
   
   ----------- end diff -----------
```

### CODE REVIEW

File: tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php
Comment (julienfalque) [inline tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php:141]: Could you add tests with comments like e.g.
```php
while (foo()) {
    // noop
}
```
In this example, the body shouldn't be considered empty IMO.

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [inline src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:37]: Can we make those constants private?

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [inline src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:142]: I think `;` is better to express that the empty loop is intentional, I would make it the default.

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [inline src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:44]: Can be a constant.

**Review by julienfalque — COMMENTED** [review]
:+1:

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (SpacePossum) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:37]: than we cannot use these in unit test, I'm fine either way

File: tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php
Comment (SpacePossum) [reply tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php:141]: why not, if you use braces as style it won't be touched, if you want to use `;` than it will be fixed and leave the comment,
if you want to use `;` but have exceptions as this one than don't use the rule

File: src/RuleSet/Sets/SymfonySet.php
Comment (SpacePossum) [reply src/RuleSet/Sets/SymfonySet.php:52]: no, I just checked the repo

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:37]: I would say it's a good thing: if we change the value of a constant, it's a BC break and the test will prevent that.

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (SpacePossum) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:44]: that would be at the cost of visibility, unless you mean something else ( https://3v4l.org/mEY0G )

File: tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php
Comment (julienfalque) [reply tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php:141]: Fair enough.

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:44]: I meant
```php
private const TOKEN_LOOP_KINDS = [T_FOR, T_FOREACH, T_WHILE];
```

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:142]: Not enough to assume they want to follow the `braces` style IMO. /cc @nicolas-grekas maybe?

Comment (nicolas-grekas) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:142]: better use `{}` I suppose since it's a bit more readable/explicit? I've no strong opinion on the topic.

File: src/RuleSet/Sets/SymfonySet.php
Comment (SpacePossum) [reply src/RuleSet/Sets/SymfonySet.php:52]: talk continues here https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5745#discussion_r640626708

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (julienfalque) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:136]: Ok, let's keep `braces` as default value then.

**Review by julienfalque — APPROVED** [review]

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (keradus) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php:136]: I would love to put one more argument. real example from our own codebase:
```diff
        for (++$index; $tokens[$index]->isWhitespace(" \t") || $tokens[$index]->isComment(); ++$index);
        for (++$index; $tokens[$index]->isWhitespace(" \t") || $tokens[$index]->isComment(); ++$index) {
        }
```

- version using `{}` - oh, we had a loop that was doing sth one day, and now we no longer execute anything inside, so the whole loop can be dropped! awesome, dead code detected, removing ;) ## ups, I broke the code
- version using `;` - ah, we are moving the index, the code is needed.

(so basically what @julienfalque said)

-----

I strongly suggest to use `;` by default.
@nicolas-grekas , can you reconsider your opinion?

@julienfalque , @SpacePossum  - I suggest to make fixer default `;` whatever gonna be decision of for Symfony ruleset

**Review by keradus — COMMENTED** [review]
let's (re-)align on https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5745#discussion_r679564664 before merging

**Comment by SpacePossum** [comment]
I've changed the default and removed the rule from the Symfony set. This way the talk about (default) value in sets can be done in new PR's with representation of the community of the set, while not blocking this PR from proceeding.

File: src/Fixer/ControlStructure/EmptyLoopBodyFixer.php
Comment (keradus) [reply src/Fixer/ControlStructure/EmptyLoopBodyFixer.php]: (ref) https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5745#issuecomment-889763849
> I've changed the default and removed the rule from the Symfony set. This way the talk about (default) value in sets can be done in new PR's with representation of the community of the set, while not blocking this PR from proceeding.

@nicolas-grekas , don't hesitate to create new ticket to update Symfony set

**Review by keradus — APPROVED** [review]

**Pull request merged**
